src/components/Axis.tsx

```diff
@@ -34,523 +36,510 @@ interface AxisState {
   calculatedLabelPosition?: number
 }
 
-class Axis extends React.Component<AxisProps, AxisState> {
-  constructor(props) {
-    super(props)
-    this.state = { hoverAnnotation: 0 }
-  }
-  axisRef?: { querySelectorAll: Function } = null
+const boundingBoxMax = (axisNode, orient) => {
+  const axisRef = axisNode.current
+  if (!axisRef) return 30
 
-  boundingBoxMax = () => {
-    // && this.props.dynamicLabel ???
-    if (!this.axisRef) return 30
-    const { orient = "left" } = this.props
-
-    const positionType =
-      orient === "left" || orient === "right" ? "width" : "height"
-
-    const axisLabelMax =
-      Math.max(
-        ...[...this.axisRef.querySelectorAll(".axis-label")]
-          .map(
-            (l: { getBBox: Function }) =>
-              (l.getBBox && l.getBBox()) || { height: 30, width: 30 }
-          )
-          .map(d => d[positionType])
-      ) + 25
-
-    return axisLabelMax
-  }
+  const positionType =
+    orient === "left" || orient === "right" ? "width" : "height"
+
+  const axisLabelMax =
+    Math.max(
+      ...[...axisRef.querySelectorAll(".axis-label")]
+        .map(
+          (l: { getBBox: Function }) =>
+            (l.getBBox && l.getBBox()) || { height: 30, width: 30 }
+        )
+        .map((d) => d[positionType])
+    ) + 25
+
+  return axisLabelMax
+}
+
+export default (props: AxisProps) => {
+  const {
+    rotate,
+    label,
+    orient = "left",
+    marginalSummaryType,
+    tickFormat = marginalSummaryType ? () => "" : (d) => d,
+    size,
+    width = (size && size[0]) || 0,
+    height = (size && size[1]) || 0,
+    className,
+    padding,
+    tickValues,
+    scale,
+    ticks,
+    footer,
+    tickSize,
+    tickLineGenerator,
+    baseline = true,
+    margin = { top: 0, bottom: 0, left: 0, right: 0 },
+    center = false,
+    annotationFunction,
+    glyphFunction,
+    xyPoints
+  } = props
+
+  const [hoverPosition, changeHoverPosition] = useState(0)
+  const [calculatedLabelPosition, changeCalculatedLabelPosition] =
+    useState(undefined)
+
+  const axisNode = useRef(null)
+
+  useEffect(() => {
+    const { label = { position: false }, dynamicLabelPosition } = props
 
-  componentDidUpdate() {
-    const { label = { position: false }, dynamicLabelPosition } = this.props
-    const { calculatedLabelPosition } = this.state
     if (!label.position && dynamicLabelPosition) {
-      const newBBMax = this.boundingBoxMax()
+      const newBBMax = boundingBoxMax(axisNode, orient)
       if (newBBMax !== calculatedLabelPosition) {
-        this.setState({ calculatedLabelPosition: newBBMax })
+        changeCalculatedLabelPosition(newBBMax)
       }
     }
-  }
+  })
```

Comment:
I guess this effect can benefit from strict dependencies list. Let's destructure `label` and `dynamicLabelPosition` earlier in the component and just use direct prop names

```js
useEffect(() => {
  if (!label.position && dynamicLabelPosition) {
    /* ... */
  }
}, [label, dynamicLabelPosition]);
```

Also, doing so I'd suggest not to default `label` to an object, this would make effect invalidate too often.

Reply:
sounds good